components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -69,29 +69,39 @@ public Policy getPolicy(String policyLevel, String policyName) throws APIMgtDAOE
                 return getApplicationPolicy(policyName);
             } else if (APIMgtConstants.ThrottlePolicyConstants.SUBSCRIPTION_LEVEL.equals(policyLevel)) {
                 return getSubscriptionPolicy(policyName);
+            } else {
+                throw new APIMgtDAOException("Couldn't find the policy for name: " + policyName + ", level: " +
+                        policyLevel);
             }
         } catch (SQLException e) {
             throw new APIMgtDAOException("Couldn't find " + policyName + ' ' + policyLevel + "Policy", e);
         }
-        return null;
     }
 
+    /**
+     *@see PolicyDAO#getPolicyByUuid(String, String)
+     */
     @Override
-    public Policy getPolicyByUuid(String uuid, String policyLevel) throws APIMgtDAOException {
+    public Policy getPolicyByUuid(String policyLevel, String uuid) throws APIMgtDAOException {
```

Comment:
IMHO if we have APIs for each specific policy type, we will have to change the API each time a different type of policy is added, which is unnecessary.